floris/simulation/flow_field.py

```diff
@@ -73,23 +73,23 @@ def _discretize_turbine_domain(self):
         Create grid points at each turbine
         """
         xt = [coord.x1 for coord in self.turbine_map.coords]
-        rotor_points = int(np.sqrt(self.turbine_map.turbines[0].grid_point_count))
-        x_grid = np.zeros((len(xt), rotor_points, rotor_points))
-        y_grid = np.zeros((len(xt), rotor_points, rotor_points))
-        z_grid = np.zeros((len(xt), rotor_points, rotor_points))
+        ngrid = self.turbine_map.turbines[0].ngrid
+        x_grid = np.zeros((len(xt), ngrid, ngrid))
+        y_grid = np.zeros((len(xt), ngrid, ngrid))
+        z_grid = np.zeros((len(xt), ngrid, ngrid))
 
         for i, (coord, turbine) in enumerate(self.turbine_map.items):
+
+            x2, x3 = coord.x2, coord.x3
+
+            # Save the indices of the flow field points for this turbine
+            turbine.flow_field_point_indices = i * ngrid * ngrid + np.arange(ngrid * ngrid)
+
+            pt = turbine.rloc * turbine.rotor_radius
+
             xt = [coord.x1 for coord in self.turbine_map.coords]
-            yt = np.linspace(
-                coord.x2 - turbine.rotor_radius,
-                coord.x2 + turbine.rotor_radius,
-                rotor_points,
-            )
-            zt = np.linspace(
-                coord.x3 - turbine.rotor_radius,
-                coord.x3 + turbine.rotor_radius,
-                rotor_points,
-            )
+            yt = np.linspace(x2 - pt, x2 + pt, ngrid,)
+            zt = np.linspace(x3 - pt, x3 + pt, ngrid,)
```

Comment:
There are a couple of extra commas here that could be removed now that these are one-liners, which is a nice touch.